### PR TITLE
multiple: rename SeedManifest to Manifest after moving into seedwriter package (2/3)

### PR DESCRIPTION
**Pull request by Meulengracht**
Renamed all references of SeedManifest to Manifest unless it made sense given the context. (I.e variable names outside seed package, docstrings, comments that refer to it as 'seed manifest')

**Parts**
1: https://github.com/snapcore/snapd/pull/12699 (move to seedwriter package)
2: https://github.com/snapcore/snapd/pull/12732 (rename to Manifest)
3: https://github.com/snapcore/snapd/pull/12700 (integration into seedwriter)
4: https://github.com/snapcore/snapd/pull/12733 (extend usage of manifest in seedwriter)

### CODE REVIEW

File: seed/seedwriter/manifest_test.go [hunk 1]
```diff
@@ -55,8 +55,8 @@ func (s *manifestSuite) writeSeedManifest(c *C, contents string) string {
 	return manifestFile
 }
 
-func (s *manifestSuite) checkManifest(c *C, manifest *seedwriter.SeedManifest, revsAllowed, revsSeeded map[string]*seedwriter.SeedManifestSnapRevision, vsAllowed, vsSeeded map[string]*seedwriter.SeedManifestValidationSet) {
-	expected := seedwriter.NewSeedManifestForTest(revsAllowed, revsSeeded, vsAllowed, vsSeeded)
+func (s *manifestSuite) checkManifest(c *C, manifest *seedwriter.Manifest, revsAllowed, revsSeeded map[string]*seedwriter.ManifestSnapRevision, vsAllowed, vsSeeded map[string]*seedwriter.ManifestValidationSet) {
+	expected := seedwriter.NewManifestForTest(revsAllowed, revsSeeded, vsAllowed, vsSeeded)
 	c.Check(manifest, DeepEquals, expected)
 }
```
Comment (ernestl) [inline seed/seedwriter/manifest_test.go:62]: Consider also renaming:
- `TestReadSeedManifestFullHappy`
- `TestReadSeedManifestParseFails`
- `TestReadSeedManifestNoFile`

Comment (Meulengracht) [reply seed/seedwriter/manifest_test.go:62]: Done!

File: seed/seedwriter/manifest.go [hunk 2]
```diff
@@ -93,27 +93,27 @@ func (vs *SeedManifestValidationSet) hasSnap(snapName string) bool {
...
+		vsAllowed:   make(map[string]*ManifestValidationSet),
+		vsSeeded:    make(map[string]*ManifestValidationSet),
 	}
 }
 
-// NewSeedManifestForTest is stricly for unit tests, do not use for non-test code.
-func NewSeedManifestForTest(revsAllowed, revsSeeded map[string]*SeedManifestSnapRevision, vsAllowed, vsSeeded map[string]*SeedManifestValidationSet) *SeedManifest {
-	osutil.MustBeTestBinary("NewSeedManifestForTest can only be used in unit tests")
+// NewManifestForTest is stricly for unit tests, do not use for non-test code.
+func NewManifestForTest(revsAllowed, revsSeeded map[string]*ManifestSnapRevision, vsAllowed, vsSeeded map[string]*ManifestValidationSet) *Manifest {
```
Comment (pedronis) [inline seed/seedwriter/manifest.go:113]: unrelated but MockManifest would have been a more typical name for this btw

**Review by pedronis — APPROVED** [review]
thanks, one unrelated passing remark

File: seed/seedwriter/manifest.go (hunk 2 above)
Comment (Meulengracht) [reply seed/seedwriter/manifest.go:113]: Ah ok! I'll rename this, thanks for the review!

**Review by ernestl — APPROVED** [review]

**Pull request merged**